<!DOCTYPE html>
<html>
<head>
<meta charset="utf8" />
<title>патч для Makefile.am</title>
</head>
<body>

<table><tr><td style="vertical-align:top;">
<h1>патч для Makefile.am</h1>
</td><td style="vertical-align:top;">
<a href="argument-list-too-long.htm">Argument list too long</a>,
<a href="index.htm">Как собирать monodevelop</a>
<br />
<a href="../howto-roll-tarballs.htm">как роллить тарболы</a>
</td></tr></table>

<a href="https://raw.githubusercontent.com/ArsenShnurkov/shnurise-tarballs-automation/master/dev-util/monodevelop/files/main-src-core-Monodevelop.Ide-Makefile.am.patch">https://raw.githubusercontent.com/ArsenShnurkov/shnurise-tarballs-automation/master/dev-util/monodevelop/files/main-src-core-Monodevelop.Ide-Makefile.am.patch</a>
<br />
<br />

mkdir -p /var/lib/tarball-scripts/ArsenShnurkov/shnurise-tarballs-automation/dev-util/monodevelop/files
<br />
<br />
основной файл, который подаётся на вход automake, это
<br />
${WORKDIR}/main/Makefile.am
<br />
файл, который надо пропатчить это
<br />
${WORKDIR}/main/src/core/MonoDevelop.Ide/Makefile.am
<br />
Оба этих файла лежат в рабочей директории ${WORKDIR}, а рабочая директория не сохраняется (по крайней мере так происходит в portage при выключенной FEATURE="keepwork")
<br />
<br />
Поэтому скопируем main/src/core/MonoDevelop.Ide/Makefile.am в директорию скриптов роллинга (переименовав в Makefile.am.orig). Новую версию положим рядом и назовём Makefile.am.new
<br />
Дифф будем делать командой
<br />
diff -u Makefile.am.orig Makefile.am.new >files/main-src-core-Monodevelop.Ide-Makefile.am.patch
<br />
<br />
Накатывать патч будем командой
<br />
patch ${WORKDIR}/main/src/core/MonoDevelop.Ide/Makefile.am ${FILESDIR}/main-src-core-Monodevelop.Ide-Makefile.am.patch
<br />
такой вариант запуска утилиты patch задокументирован в man patch:
<br />
<a href="http://linux.die.net/man/1/patch">http://linux.die.net/man/1/patch</a>
<br />
<br />
вызов configure сам потом сгенерирует makefile.in и Makefile, так что к моменту вызова make dist 
<br />
в Makefile уже будет находится код из патча.

</body>
</html>
